### ISSUE TITLE
feat(backends): Integrate jax-md mm_forcefields as MM backend (WSL2 initially)

### ISSUE THREAD

**Issue by ericchansen** [ericchansen/q2mm#93]
## Summary

Integrate [jax-md](https://github.com/jax-md/jax-md) as an MM backend, replacing or complementing our hand-rolled JAX energy functions in `JaxEngine`. jax-md already has a production-quality `mm_forcefields/oplsaa/` module with topology builders, OPLSAA energy functions, neighbor lists, and multiple Coulomb handlers (cutoff, Ewald, PME). It also has a `mm_forcefields/reaxff/` module.

## Motivation

In #68 we wrote ~100 lines of pure-JAX energy functions (harmonic bond/angle, Fourier torsion, 12-6 LJ). This was necessary because `pip install jax-md` fails on Windows due to:

```
jax-md -> flax -> orbax-checkpoint -> uvloop (Unix-only)
```

However, jax-md's `mm_forcefields` module is a superset of what we wrote -- it includes:
- OPLSAA energy assembly with bonded terms + LJ + Coulomb
- Topology auto-builder with exclusion/1-4 scaling masks
- Multiple electrostatics handlers (CutoffCoulomb, EwaldCoulomb, PMECoulomb)
- ReaxFF implementation
- Proper neighbor list management via `jax_md.partition`
- Numerically stable geometry primitives (`safe_norm`, `safe_arccos`)

We should use the battle-tested upstream code rather than maintaining our own.

## Approach

**Phase 1: WSL2 on Windows**
- Add a `[jax-md]` optional extra that installs `jax-md`
- Create a `JaxMDEngine(MMEngine)` that wraps `jax_md.mm_forcefields.oplsaa`
- Map our `ForceField` param vector to jax-md's `BondedParameters` / `NonbondedParameters`
- Use `jax.value_and_grad` on jax-md's energy function for analytical gradients
- Test under WSL2 initially (jax-md installs cleanly on Linux)
- CI via Docker (#88) or GitHub Actions Linux runners

**Phase 2: Cross-platform** (see #93, #94)
- Blocked on either Winloop support (#93) or modular jax-md deps (#94)
- Once resolved, `JaxMDEngine` becomes cross-platform

## Key jax-md APIs to use

```python
from jax_md.mm_forcefields import oplsaa

# Build topology + params from our ForceField object
energy_fn, neighbor_fn, displacement_fn = oplsaa.energy(
    topology, params, box=box, coulomb=coulomb, nb_options=nb_options
)

# Evaluate with analytical gradients
energies = energy_fn(positions, nlist)  # returns dict with bond, angle, torsion, lj, coulomb, total
grad_fn = jax.grad(lambda pos, nlist: energy_fn(pos, nlist)["total"])
```

## Relationship to existing work

- Our `JaxEngine` (#68) provides a lightweight fallback for Windows/no-WSL environments
- `JaxMDEngine` would be the preferred backend when jax-md is available
- Both share the same `MMEngine` interface, so they're interchangeable

## Acceptance criteria

- [ ] `JaxMDEngine` passes the same integration tests as `JaxEngine`
- [ ] Analytical gradients match finite differences to < 1e-4
- [ ] Works under WSL2 on Windows
- [ ] Works natively on Linux/macOS
- [ ] Documentation with WSL2 setup instructions
